src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -213,14 +216,18 @@ private static String detectRegression(String thresholdFilePath, Map<String, Big
                     for (String keyWord : scoreMap.keySet()) {
                         IonDecimal thresholdScore = (IonDecimal) thresholds.get(keyWord);
                         if (scoreMap.get(keyWord).compareTo(thresholdScore.bigDecimalValue()) < 0) {
-                            String regressionDetect = "The performance regression detected when benchmark the ion-java from the new commit with the test data: " + comparisonResultStruct.get(INPUT) + " and parameters: " + comparisonResultStruct.get(PARAMETERS);
-                            return regressionDetect;
+                            regressions.put(keyWord, scoreMap.get(keyWord));
                         }
                     }
                 }
             }
+            if (regressions.size() != 0) {
+                return "The performance regression detected when benchmark the ion-java from the new commit with the test data: "
+                        + fileName.substring(fileName.lastIndexOf(File.separator) + 1) + " and parameters: " + comparisonResultStruct.get(PARAMETERS) + System.lineSeparator()
+                        + "The following aspects have regressions: " + regressions + System.lineSeparator();
+            } else {
+                return System.lineSeparator();
```

Comment:
Is returning the empty string `""` sufficient?

Reply:
> sufficient

Return the empty string "" can make sure that only regression detection messages will be printed in the workflow which will be easier for users to tract the regression detected message. Because there is extra empty line in the workflow, 'if condition' will be added in the following commit to avoid printing empty line.